src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -202,6 +202,22 @@ public IWebHost Build()
                 host.Dispose();
                 throw;
             }
+
+            IServiceProvider GetProviderFromFactory(IServiceCollection collection)
```

Comment:
Only used once and not complex, inline.

Reply:
I think it should remain a method. It's not like it's 2 lines of code. It's a few.